### PR TITLE
Underrated logic change

### LINKED ISSUE #61: Improve logic of underrated endpoints
Currently, we are sorting on the `custom_popularity` column while fetching the data from the DB. However, this is not producing the desired results. We need to display songs that are underrated _and_ recently posted.
I imagine a modification to the logic as follows:

1. We decide upon certain lower and upper thresholds for which a song maybe decided to be underrated. These thresholds maybe applied on `custom_popularity` or more conveniently on YouTube views since we are currently supporting only YouTube links. For example `views < 300000` and `views > 500`
2. We make a join on `user_posts` and `link` with the above conditions and sort the resulting posts by date.

### PR DESCRIPTION
**Pull request by thescriptninja**
This PR aims to improve on the existing logic for serving 'underrated' songs. Earlier, the songs that were served were ordered by `custom_popularity`. This caused extremely obscure songs to appear first in order. However, the expected behavior (as I imagine it to be) is to serve the songs that were recently posted _and_ are underrated or are not as popular.
This is solved by placing a upper threshold on `custom_popularity`. I have set an arbitrary value of `0.5` that I observed while testing some links manually against their corresponding `custom_popularity` score. The posts are then sorted by date

Closes #61

### CODE REVIEW

File: c3po/db/models/link.py [hunk 1]
```diff
@@ -16,10 +16,14 @@ class Link(Base):
 
     post_count = Column("post_count", Integer)
     likes_count = Column("likes_count", Integer)
+    views = Column("views", BigInteger)
```
Comment (mukul-mehta) [inline c3po/db/models/link.py:19]: Using BigInteger for views so that Gangnam style doesn't break your system 😆

Comment (thescriptninja) [reply c3po/db/models/link.py:19]: Exactly :laughing:

**Comment by thescriptninja** [comment]
> I think it might be a good idea to fetch the threshold value from the `config.ini` file instead of putting the value here directly

@mukul-mehta Done :v:

**Comment by thescriptninja** [comment]
One thing that is unfortunately happening is that since some people have posted Hindi (T-series :roll_eyes:) songs from around 2010-12 and they have a lot of views, those videos are falling in this bracket, since `custom_popularity` falls below the threshold. We can add some form of filtering based on views to this too.

File: c3po/job/utils/metadata.py [hunk 2]
```diff
@@ -26,35 +28,38 @@ def insert_metadata(raw_data):
...
+                    if(data.track and data.artists):
+                        new_song = _insert_song(data.track, session)
+                        new_link.song_id = new_song.id
+                        for artist_data in data.artists:
+                            new_artist = _insert_artist(artist_data, session)
+                            _insert_artist_song(new_artist, new_song, session)
+                            _insert_song_genre(artist_data, new_song, session)
+            except ServerNotFoundError:
+                print("Google API unreachable!")
+                time.sleep(30)
```
Comment (xypnox) [inline c3po/job/utils/metadata.py:42]: :roll_eyes:

**Comment by xypnox** [comment]
I took `underrated logic change` literally. That is: Logic change that is underrated :stuck_out_tongue:

File: c3po/job/utils/metadata.py (hunk 2 above)
Comment (thescriptninja) [reply c3po/job/utils/metadata.py:42]: The API times out sometimes :confused: 
I thought putting it to sleep temporarily and checking again for the next entry should work :sweat_smile:

**Pull request merged**